ksql-engine/src/test/java/io/confluent/ksql/physical/PhysicalPlanBuilderTest.java

```diff
@@ -449,15 +441,34 @@ private void shouldUseProvidedOptimizationConfig(Object value) {
   }
 
   @Test
-  public void shouldUseOptimizationConfigProvidedWhenOn() throws Exception {
+  public void shouldUseOptimizationConfigProvidedWhenOn() {
     shouldUseProvidedOptimizationConfig(StreamsConfig.OPTIMIZE);
   }
 
   @Test
-  public void shouldUseOptimizationConfigProvidedWhenOff() throws Exception {
+  public void shouldUseOptimizationConfigProvidedWhenOff() {
     shouldUseProvidedOptimizationConfig(StreamsConfig.NO_OPTIMIZATION);
   }
 
+  @Test
+  public void shouldConfigureProductionExceptionHandlerName() {
+    // Given:
+    final OutputNode spyNode = spy(
```

Comment:
spies are usually a code smell (see the warning against them in javadoc for `Mockito#spy`) - instead can we move `TestOutputNode` to its own class and use it?

Reply:
Discussed offline with @agavra : this spy is difficult to get around since the OutputNode is a KsqlBareOutputNode, which generates a random queryId (ignoring the queryIdGenerator passed in). Leaving as is for now, though @agavra hopes to update this flow (and remove the need for this spy) as part of his upcoming work that will involve a different mechanism for generating query IDs.